libqtile/core/manager.py

```diff
@@ -422,13 +424,19 @@ def grab_chord(self, chord: KeyChord) -> None:
             hook.fire("enter_chord", chord.mode)
 
         self.ungrab_keys()
+        self.core.grab_keyboard()
         for key in chord.submappings:
-            self.grab_key(key)
+            # TODO figure out how to do this without reaching into backend
+            # internals
+            keysym, mask_key = self.core.lookup_key(key)
+            mask_key &= self.core._valid_mask
+            self.keys_map[(keysym, mask_key)] = key
 
     def cmd_ungrab_chord(self) -> None:
         """Leave a chord mode"""
         hook.fire("leave_chord")
 
+        self.core.ungrab_keyboard()
         self.ungrab_keys()
```

Comment:
It would probably be best to add `ungrab_keyboard()` and `grab_keyboard()` to the `Qtile` manager class. Then `ungrab_keyboard` could call the `self.core` implementation and do a `self.keys_map.clear()`. That would be the only part of `self.ungrab_keys()` that is necessary. Thus the two lines would become just the one 
```suggestion
        self.ungrab_keyboard()
```

Reply:
In light of my last commit, I don't think this is true -- since `grab_chord` now grabs keys as well as grabbing the keyboard, I think calling `ungrab_keys` is necessary as well.

Of course, we could still create an `ungrab_keyboard` on the manager class and make it call `ungrab_keys` as well.

Reply:
Yeah, I think technically it would be sufficient to populate the `keys_map` with the right values without additional single-key grabs in the backend. But then `Qtile.grab_key` populates the `keys_map` from the output of `core.grab_key`.
One could add a parameter to the `core.grab_key` to just return the dictionary key without doing the actual backend call, but I'm fine with the way you did it. It seems a bit unintuitive to grab the whole keyboard and some keys at first sight but then it's really not worth the extra hassle.
Can you mark this as resolved?